.github/workflows/ci-model-regression.yml

```diff
@@ -388,7 +388,7 @@ jobs:
         with:
           path: ~/.cache/pypoetry/virtualenvs
           key: ${{ runner.os }}-poetry-3.8-${{ hashFiles('**/poetry.lock') }}
-          restore-keys: ${{ runner.os }}-poetry-3.8
+          restore-keys: ${{ runner.os }}-poetry-3.8-${{ hashFiles('**/poetry.lock') }}
```

Comment:
[from the action's README](https://github.com/actions/cache#inputs):
> `restore-keys` - An ordered list of keys to use for restoring the cache if no cache hit occurred for key

So it sounds like we may want to remove the option altogether?

Reply:
No, we want it. The issue that we're turning to solve here is that cache is shared trough workflows, jobs, and so on. Currently, we use ${{ runner.os }}-poetry-3.8 which means that any cache that matches a restore key is used. It might happen that the cache will be overridden by different PR that why we want to be sure that we use cache related to our dependencies.

Does it make sense?

Reply:
Good point!